arrays/index.js

```diff
@@ -0,0 +1,170 @@
+//Створи масив «Список покупок». Кожен елемент масиву є об'єктом, який
+//містить назву продукту, кількість і куплений він чи ні, ціну за одиницю
+// товару, сума. Написати кілька функцій для роботи з таким масивом:
+//Виводити весь список на екран таким чином, щоб спочатку йшли продукти,
+//що ще не придбані, а потім - ті, що вже придбали.
+//Покупка продукту. Функція приймає назву продукту і відзначає його
+// як придбаний.
+const myList = [
+  { name: "chicken", quantity: 2, purchased: true, price: 110.5 },
+  { name: "lemon", quantity: 0.5, purchased: true, price: 67.9 },
+  { name: "apple", quantity: 6, purchased: false, price: 12 },
+  { name: "rice", quantity: 1, purchased: true, price: 70 },
+  { name: "cola", quantity: 3, purchased: false, price: 55 },
+  { name: "water", quantity: 10, purchased: false, price: 17.5 },
+  { name: "bread", quantity: 1, purchased: false, price: 25 },
+  { name: "cake", quantity: 1, purchased: false, price: 155 },
+];
+const listWithTotal = myList.map((listItem) => ({
+  ...listItem,
+  total: listItem.quantity * listItem.price,
+}));
+console.log(listWithTotal);
+
+function sortMyList(arr) {
+  return [...arr].sort((a, b) => a.purchased - b.purchased);
+}
+
+function markAsPurchased(arr, product) {
+  const neededIndex = arr.findIndex((arrItem) => arrItem.name == product);
+  if (neededIndex === -1) {
+    return "No such product";
+  } else {
+    arr[neededIndex].purchased = true;
+    return arr;
+  }
+}
+console.log(markAsPurchased(myList, "bread"));
+console.log(sortMyList(myList));
+
+//Видалення продукту зі списку (видалення повинно проводитися шляхом
+//створення нового масиву, в якому продукт, що ми шукаємо, буде відсутнім)
+
+function removeProduct(arr, prodToRemove) {
+  const reserveArr = [...arr];
+  let arrWithoutProduct = [];
+  if (reserveArr.findIndex((item) => item.name === prodToRemove) === -1) {
+    return "The product is not found in the list";
+  } else {
+    arrWithoutProduct = reserveArr
+      .filter((item) => item.name !== prodToRemove)
```

Comment:
method filter will creat new array there is no need for const reserveArr = [...arr];
  let arrWithoutProduct = [];